first_analysis.py

```diff
@@ -33,3 +33,23 @@
 display(mod.summary())
 
 # OK, we need some more analysis now.
+
+# adding a fancy plot
+
+df['gender_dummy'] = df['gender'].replace(['male', 'female'], [0,1])
+survived = df[df['survived_dummy'] == 1]
+died = df[df['survived_dummy'] == 0]
+
+fig2 = plt.figure(figsize = (12, 8))
+ax2 = fig2.add_subplot(projection='3d')
+ax2.scatter(survived['gender_dummy'], survived['fare'],survived['survived_dummy'],
+           color = 'red', label = 'survived')
```

Comment:
I wonder whether this might be easier to read?

```suggestion
ax2.scatter(
    survived['gender_dummy'],
    survived['fare'],
    survived['survived_dummy'],
    color = 'red',
    label = 'survived')
```